src/facade/dragonfly_connection.cc

```diff
@@ -1032,15 +1053,6 @@ void Connection::DispatchFiber(util::FiberSocketBase* peer) {
 
   DCHECK(cc_->conn_closing || builder->GetError());
   cc_->conn_closing = true;
-
```

Comment:
I placed it right after the PRECLOSE phase flag because it's after that point we can't receive any more messages. It was _correct_ before, but somewhat fragile

The troublesome part specifically is:
- We are dispatching a command
- We were asked to close, so `conn_closing` is true
- Control messages can be still sent on conn_closing, because we might be tracking the dispatch with a client pause, takeover, etc.
- They are only sent when a dispatch is ongoing, i.e. when the dispatch fiber is not yet joined, or we're stuck on a sync dispatch, i.e. we don't wait for the dispatch fiber's join yet
- But if we invent any new control message (like quit instantly with client kill for example), we might run into trouble that it'll be never cleaned up because the dispatch fiber already joined